Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -369,28 +369,33 @@ int RGroupDecomposition::add(const ROMol &inmol) {
     return -2;
   }
 
+  const bool prune = true;
+  // in case the value ends up being changed in a future version of the code:
+  if (prune) {
+    data->permutationProduct = 1;
+  }
   if (data->params.matchingStrategy != GA) {
-    size_t N = 1;
-    for (auto &matche : data->matches) {
-      size_t sz = matche.size();
+    size_t N = data->permutationProduct;
+    for (auto matche = data->matches.begin() + data->previousMatchSize;
+         matche != data->matches.end(); ++matche) {
+      size_t sz = matche->size();
       N *= sz;
     }
     // oops, exponential is a pain
     if (N * potentialMatches.size() > 100000) {
-      data->permutation = std::vector<size_t>(data->matches.size(), 0);
-      data->process(true);
+      data->permutationProduct = N;
```

Comment:
Have we tried an example where this is the case?  I know the older behavior, but not the new one.  we didn't have a real test for this before except for running it on larger datasets,